File: Python/0701-0750/713. Subarray Product Less Than K.py
# https://leetcode.com/problems/subarray-product-less-than-k/

class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k <= 1: # No Possible Subarray found
            return 0

        left, right, product, result = 0, 0, 1, 0
        size = len(nums)

        while right < size:
            product *= nums[right]
            
            # Shrink the window till the product is less than or equal to K
            while product >= k:
                product /= nums[left]
                left += 1

            result += 1 + (right - left) #Update result
            right += 1 # Move right

        return result

File: Java/0701-0750/713. Subarray Product Less Than K.java
// https://leetcode.com/problems/subarray-product-less-than-k/

class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        if (k <= 1) return 0; // No Possible Subarray found

        int left = 0, right = 0, product = 1, result = 0;
        int size = nums.length;

        while(right < size){
            product *= nums[right];
            
            // Shrink the window till the product is less than or equal to K
            while (product >= k) product /= nums[left++];

            result += 1 + (right - left); // Update result
            right++; // Move right
        }

        return result;
    }
}
